### PR TITLE
Issue #3257: Allowing color elements to be put in multiple fieldsets.

### PR DESCRIPTION
**Pull request by hosef**
Moved the code to create the render arrays for color elements into a function that can be called from the theme settings page and changed the code in the color module to use the function as well, converted the JS code to use data attributes for finding the color elements, and fixed some bugs and coding standards problems.

### CODE REVIEW

**Comment by jenlampton** [comment]
new sandbox...

**Comment by jenlampton** [comment]
Sandboxme?

File: core/modules/color/color.module [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Comment (jenlampton) [inline core/modules/color/color.module]: Can we remove this extra space? (and maybe if your editor added it - you can configure it not to, as we don't like these extra lines in Backdrop)

File: core/themes/basis/theme-settings.php [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Comment (jenlampton) [inline core/themes/basis/theme-settings.php]: extra line here should be removed

File: core/themes/basis/theme-settings.php [hunk 3]
```diff
@@ -7,6 +8,51 @@
  * inform the user that the module supports color schemes if the Color module
  * is enabled.
  */
-$form['color'] = array(
-  '#markup' => '<p>' . t('This theme supports custom color palettes if the Color module is enabled on the <a href="!url">modules page</a>. Enable the Color module to customize this theme.', array('!url' => url('admin/modules'))) . '</p>',
-);
+
```
Comment (jenlampton) [inline core/themes/basis/theme-settings.php]: extra line?

File: core/themes/basis/theme-settings.php [hunk 4]
```diff
@@ -7,6 +8,51 @@
...
+  );
+  $fields = array(
+    'header',
+    'base',
+    'slogan',
+    'titleslogan',
+    'hovermenu',
+  );
+  foreach ($fields as $field) {
+    $form['header'][$field] = color_get_color_element('basis', $field, $form);
```
Comment (jenlampton) [inline core/themes/basis/theme-settings.php:24]: love this, nice and clean.

File: core/modules/color/color.module (hunk 1 above)
Comment (oadaeh) [reply core/modules/color/color.module]: I think if that's a big enough deal to hold up a PR, it should be documented here: https://api.backdropcms.org/php-standards#indenting as it is on d.o.

Comment (herbdool) [reply core/modules/color/color.module]: Just took a look https://www.drupal.org/docs/develop/standards/coding-standards#indenting - I guess Drupal does like the blank line.

Comment (jenlampton) [reply core/modules/color/color.module]: I think it should be documented even if it's not a big enough deal to hold up a pr, so https://api.backdropcms.org/php-standards :)

**Comment by hosef** [comment]
Reopening to force tests to be re-run

**Comment by herbdool** [comment]
I'm going to try restart the site.

**Comment by quicksketch** [comment]
Thanks @hosef! Finally merged into 1.x and 1.12.x.

**Pull request merged**